### PR TITLE
Custom AEM Content Fragment Editor RTE Toolbar Button

### PR DESCRIPTION
**Pull request by slitviachenko**

### CODE REVIEW

File: cf-editor-rte-toolbar-button-sample/extension-manifest.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "CF Editor Rich Text Editor Toolbar Button",
+  "id": "cf-editor-rich-text-editor-toolbar-button",
+  "description": "Enhance your CF Editor experience by seamlessly adding a Rich Text Editor toolbar button for expanded functionality.",
```
Comment (lenaorobei) [inline cf-editor-rte-toolbar-button-sample/extension-manifest.json:4]:
```suggestion
  "description": "Sample extension showcasing the extensibility of the Rich Text Editor Toolbar.",
```

File: cf-editor-rte-toolbar-button-sample/extension-manifest.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  "version": "0.0.1",
+  "rte": {
+    "toolbarButtons": [
+      {
+        "id": "aem-extensibility-cf-editor-rte-toolbar-button",
+        "tooltip": "Explore expanded CF Editor functionality with the addition of a Rich Text Editor toolbar button for enhanced text customization."
```
Comment (lenaorobei) [inline cf-editor-rte-toolbar-button-sample/extension-manifest.json:10]:
```suggestion
        "tooltip": "Click to explore the extensibility features"
```

File: cf-editor-rte-toolbar-button-sample/package.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    "@babel/plugin-transform-react-jsx": "^7.8.3",
+    "@babel/polyfill": "^7.8.7",
+    "@babel/preset-env": "^7.8.7",
+    "@openwhisk/wskdebug": "^1.3.0"
+  },
+  "scripts": {
+    "test": "jest --passWithNoTests ./test",
+    "e2e": "jest --collectCoverage=false --testRegex ./e2e"
+  },
+  "description": "Enhance your CF Editor experience by seamlessly adding a Rich Text Editor toolbar button for expanded functionality.",
```
Comment (lenaorobei) [inline cf-editor-rte-toolbar-button-sample/package.json:35]:
```suggestion
  "description": "Sample extension showcasing the extensibility of the Rich Text Editor Toolbar.",
```

File: cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/components/App.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
+/*
```
Comment (lenaorobei) [inline cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/components/App.js:1]: Pls add license header

File: cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/components/ExtensionRegistration.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  }
+  const init = async () => {
+    const guestConnection = await register({
+      id: extensionId,
+      methods: {
+        rte: {
+          getCustomButtons: () => ([
+            // RTE Toolbar custom button
+            {
+              id: "aem-extensibility-cf-editor-rte-toolbar-button",
```
Comment (lenaorobei) [inline cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/components/ExtensionRegistration.js:48]: Was this ID generated by the template?
If yes, let's come up with something more distinct and unique, so we can use that ID for the user guides.

File: cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/components/ExtensionRegistration.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+              tooltip: "Unlock the power of extensibility in the Content Fragment Editor - explore enhanced text customization with a custom toolbar button for the Rich Text Editor",
```
Comment (lenaorobei) [inline cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/components/ExtensionRegistration.js:49]:
```suggestion
              tooltip: "Click to explore the extensibility features",
```

File: cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/index.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+  const mockIms = {}
+
+  // render the actual react application and pass along the runtime object to make it available to the App
+  ReactDOM.render(
+    <App runtime={mockRuntime} ims={mockIms} />,
+    document.getElementById('root')
+  )
+}
+
+function bootstrapInExcShell () {
```
Comment (lenaorobei) [inline cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/index.js:39]: Do we need this method?

File: cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/index.js [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+  require('./exc-runtime')
+  // if there are no errors, bootstrap the app in the Experience Cloud Shell
+  init(bootstrapInExcShell)
+} catch (e) {
+  console.log('application not running in Adobe Experience Cloud Shell')
+  // fallback mode, run the application without the Experience Cloud Runtime
+  bootstrapRaw()
+}
+
+function bootstrapRaw () {
```
Comment (lenaorobei) [inline cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/index.js:27]: Do we need this method?

File: cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/utils.js [hunk 9]
```diff
@@ -0,0 +1,57 @@
+/*
```
Comment (lenaorobei) [inline cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/utils.js:1]: Do we need this file?

**Review by lenaorobei — CHANGES_REQUESTED** [review]
Lood good 💯 

Please address some minor comments:
1. Missing license headers
2. Redundant methods/files
3. Tooltips

Thanks!

File: cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/components/ExtensionRegistration.js (hunk 5 above)
Comment (slitviachenko) [reply cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/components/ExtensionRegistration.js:48]: As an option we can use a repo name, `aem-uix-examples`, + a folder name, `cf-editor-rte-toolbar-button-sample`, so the id will be `aem-uix-examples-cf-editor-rte-toolbar-button-sample`. What do you think?

Comment (slitviachenko) [reply cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/components/ExtensionRegistration.js:48]: or just `aem-uix-examples-cf-editor-rte-toolbar-button`

**Comment by slitviachenko** [comment]
@lenaorobei , Regarding the files and helper methods, as per our conversation, they were added by the `@adobe/aem-cf-editor-ui-ext-tpl` template, which we promote as a starter template for generating CF Editor extensions. It seems like we need to fix it, and once that's done, we can re-generate this sample. I'm hesitant to modify any of those files because I'm unsure if other `aio` cli commands, essential for modifying existing App Builder projects, will function correctly without them. This includes various `aio app add ...` commands.

File: cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/index.js (hunk 8 above)
Comment (slitviachenko) [reply cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/index.js:34]: Please look at [this comment](https://github.com/adobe/aem-uix-examples/pull/7#issuecomment-1863455698)

File: cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/index.js (hunk 7 above)
Comment (slitviachenko) [reply cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/index.js:46]: Please look at [this comment](https://github.com/adobe/aem-uix-examples/pull/7#issuecomment-1863455698)

File: cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/utils.js (hunk 9 above)
Comment (slitviachenko) [reply cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/utils.js:1]: Please look at [this comment](https://github.com/adobe/aem-uix-examples/pull/7#issuecomment-1863455698)

**Comment by slitviachenko** [comment]
@lenaorobei , I went ahead and removed unnecessary dependencies, as I'm unsure if fixing a base template is feasible in the near future.

File: cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/components/ExtensionRegistration.js (hunk 5 above)
Comment (lenaorobei) [reply cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/components/ExtensionRegistration.js:55]: Yes, `aem-uix-examples-cf-editor-rte-toolbar-button` will work.

File: cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/index.css [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+
+html,
+body {
+  margin: 0;
```
Comment (lenaorobei) [inline cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/index.css:15]: Is this file needed?

Comment (slitviachenko) [reply cf-editor-rte-toolbar-button-sample/src/aem-cf-editor-1/web-src/src/index.css:15]: It is in use, but apparently, it is not needed. I'll remove it.

**Review by lenaorobei — APPROVED** [review]
